### PR TITLE
trailing whitespaces and long lines breaks typical pre-commit hooks

### PR DESCRIPTION
**Pull request by daniel-mohr**
cf. https://github.com/hermes-hmc/ci-templates/issues/5

This merge request provides running pre-commit as github action to ensure typical coding style is followed.

This ensures that user of the templates do not get errors in their ci pipelines.

Further the already existing files are adapted to this common coding style.

### CODE REVIEW

**Comment by daniel-mohr** [comment]
This github action may also be interesting to avoid https://github.com/hermes-hmc/workflow/issues/200.

File: README.md [hunk 1]
```diff
@@ -1,18 +1,23 @@
...
+- Continuous integration templates are licensed under
+  [CC0-1.0](https://github.com/hermes-hmc/ci-templates/blob/main/LICENSES/CC0-1.0.txt).
 
 Please see the [LICENSES/](LICENSES/) directory and the headers in files for details.
 
-Copyright (c) 2023 German Aerospace Center (DLR), Forschungszentrum Jülich, Helmholtz-Zentrum Dresden-Rossendorf
-
+Copyright (c) 2023 German Aerospace Center (DLR),
+                   Forschungszentrum Jülich,
+                   Helmholtz-Zentrum Dresden-Rossendorf
```
Comment (led02) [inline README.md:23]: **nitpick:** I'm not a big fan of synthetic alignment (it has not impact on the rendered result), plus I see no actual reason to break this line up as it is way less than 120 characters. However, this is something that is most probably highly subjective... and in relation to all the very valuable changes you made, I can most probably live with this... :)

File: README.md [hunk 2]
```diff
@@ -1,18 +1,23 @@
...
 
-This repository contains templates for continuous integration systems that can be used to publish software automatically.
+This repository contains templates for continuous integration systems
+that can be used to publish software automatically.
 
 ## License
 
-- Continuous integration templates are licensed under [CC0-1.0](https://github.com/hermes-hmc/ci-templates/blob/main/LICENSES/CC0-1.0.txt).
+- Continuous integration templates are licensed under
+  [CC0-1.0](https://github.com/hermes-hmc/ci-templates/blob/main/LICENSES/CC0-1.0.txt).
```
Comment (led02) [inline README.md:17]: **though:** We should think about the actual refernece for this link. Looks kind of shaky for me....

**Review by led02 — APPROVED** [review]
Thank you for your well designed adaptations and also for testing our templates so thoroughly. Your contribution is much appreciated!

File: README.md (hunk 1 above)
Comment (daniel-mohr) [reply README.md:23]: Yes, this is not about rendering. As far as I understand markdown should be readable without rendering. And therefore typical tools like [markdownlint](https://github.com/DavidAnson/markdownlint)  claims [MD013 - Line length](https://github.com/DavidAnson/markdownlint/blob/main/doc/md013.md): `README.md:2:81 MD013/line-length Line length [Expected: 80; Actual: 122]`.

Maybe I did not explain my proceeding. I wanted just use your workflow/pipelines in my projects. Therefore I tested it with [pfu](https://gitlab.com/dlr-pa/pfu). And this project already has a ci pipeline using pre-commit. And this pipelines breaks using/integrating your pipelines.

Further I expect many people are using pre-commit and other tools with default configuration. So using a larger line-length is no option.

In the end I tried to make it readable as text and also pass the typical rules.

File: README.md (hunk 2 above)
Comment (daniel-mohr) [reply README.md:17]: The same here. I tried to make it readable as text and also pass the typical rules. The original line was about 139 characters and this is too much.

Comment (led02) [reply README.md:17]: I totally agree with the line length here. However, I do not like the reference of the link too much...

**Pull request merged**